htdocs/core/modules/DolibarrModules.class.php

```diff
@@ -2275,4 +2275,287 @@ public function remove($options = '')
     {
         return $this->_remove(array(), $options);
     }
+
+    /**
+     * Create tables and keys required by module.
+     * Files module.sql and module.key.sql with create table and create keys
+     * commands must be stored in directory reldir='/module/sql/'
+     * This function is called by this->init
+     *
+     * @param  string $reldir Relative directory where to scan files
+     * @return int             <=0 if KO, >0 if OK
+     */
+    function _load_demo_data($reldir)
```

Comment:
When does the function_load_demo_data will be called. From which screen ?

Reply:
I could not  integrate it in modulebuilder.

The function may be called through a button from the configuration file of any module by :

if(empty($conf->global->MYMODULE_DEMODATALOADED)) print '<a class="butAction" href="' .$_SERVER['PHP_SELF'] . "?action=loaddemodata" . '">' . $langs->trans('loaddemodata') . '</a>'."\n";
else print '<a class="butAction" href="' .$_SERVER['PHP_SELF'] . "?action=unloaddemodata" . '">' . $langs->trans('unloaddemodata') . '</a>'."\n";

and then executed by :

if($action=='loaddemodata'){
    $mod->_load_demo_data('/mymodule/demodata/');
}
elseif($action=='unloaddemodata'){
    $mod->_unload_demo_data('/mymodule/demodata/');
}